crates/router/src/core/payments/operations/payment_create.rs

```diff
@@ -65,6 +66,96 @@ impl<F: Send + Clone> GetTracker<F, PaymentData<F>, api::PaymentsRequest> for Pa
             .get_payment_intent_id()
             .change_context(errors::ApiErrorResponse::PaymentNotFound)?;
 
+        let (payment_link, payment_link_id) = if let Some(payment_link_object) =
+            &request.payment_link_object
+        {
+            if !request.confirm.unwrap_or(false) {
+                let created_at @ last_modified_at = Some(common_utils::date_time::now());
+                if created_at > payment_link_object.link_expiry {
+                    return Err(report!(errors::ApiErrorResponse::InvalidRequestData {
+                        message: "link_expiry time cannot be less than current time".to_string(),
+                    }));
+                }
```

Comment:
If this validation is not dependent on any of the database object, add this to `validate_request` function